horovod/run/elastic/driver.py

```diff
@@ -178,6 +178,11 @@ def _notify_workers_host_changes(self):
 
         timestamp = _epoch_time_s()
         for (host, slot), client in self._worker_clients.items():
```

Comment:
can the elastic driver keep a reference to the `rank = 0` worker client to avoid this iteration?

Reply:
Fixed.  Added rank_assignments to index directly into worker clients.  Let me know what you think.

Reply:
Nice fix! Maybe `slot_info` should be renamed to `coordinator_slot_info`.

Reply:
Renamed.